### PR TITLE
go/vt/sqlparser: Initial support for CREATE VIEW and DROP VIEW statements.

### PR DESCRIPTION
**Pull request by reltuk**

### CODE REVIEW

**Comment by reltuk** [comment]
I spent a silly amount of time trying to root cause leaving a `ddl_skip_to_end` on the end to skip of this cruft in the mysql documentation: `[WITH [CASCADED | LOCAL] CHECK OPTION]`, but I'm gonna do that explicitly instead if/when we want/need.

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -565,13 +575,13 @@ create_statement:
...
     $$ = &DDL{Action: AlterStr, Table: $7}
   }
-| CREATE VIEW table_name ddl_skip_to_end
+| CREATE VIEW table_name AS select_statement
   {
-    $$ = &DDL{Action: CreateStr, Table: $3.ToViewName()}
+    $$ = &DDL{Action: CreateStr, View: $3.ToViewName(), ViewExpr: $5}
   }
-| CREATE OR REPLACE VIEW table_name ddl_skip_to_end
+| CREATE OR REPLACE VIEW table_name AS select_statement
```
Comment (zachmu) [inline go/vt/sqlparser/sql.y]: Too much to ask you to add a or_replace_opt matcher?

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -1486,13 +1496,13 @@ drop_statement:
...
   }
-| DROP VIEW exists_opt table_name ddl_skip_to_end
+| DROP VIEW exists_opt view_name_list
   {
     var exists bool
         if $3 != 0 {
           exists = true
         }
-    $$ = &DDL{Action: DropStr, FromTables: TableNames{$4.ToViewName()}, IfExists: exists}
+    $$ = &DDL{Action: DropStr, FromTables: $4, IfExists: exists}
```
Comment (zachmu) [inline go/vt/sqlparser/sql.y]: Should be FromViews?

File: go/vt/sqlparser/sql.y (hunk 1 above)
Comment (reltuk) [reply go/vt/sqlparser/sql.y]: This ended up needing to touch the `constraint_opt` production, but I think the change there is a strict improvement for our use case. TAL.

File: go/vt/sqlparser/sql.y (hunk 2 above)
Comment (reltuk) [reply go/vt/sqlparser/sql.y]: Thanks! The fact that this now works is also reflected in tests :).

**Pull request merged**